### DISCUSSION TITLE
Why pythonw.exe is required for Poetry on Windows?

### DISCUSSION THREAD

**Discussion by hemnstill** [python-poetry/poetry#6630] [Q&A] [answered]
Poetry  1.2.1, Python 3.10.5

Im using portable version of python that does not have pythonw.exe and all working fine.

Poetry fails on command `poetry install` but creates corrupted folder `.venv`:
`No such file or directory: 'D:\\Portable\\cpython-3.10.5-windows-msvc\\pythonw.exe'`

When the command is called again another fails with message^

```
Command D:\Python\python.exe -W ignore - errored with the following return code 3221225781, and output: 
input was : import sys

if hasattr(sys, "real_prefix"):
    print(sys.real_prefix)
elif hasattr(sys, "base_prefix"):
    print(sys.base_prefix)
else:
    print(sys.prefix)
```

Ideally that Poetry not require pythonw.exe

Would be great to handle error more accurately.

**Comment by neersighted** [comment, marked as answer]
Poetry creates virtual environments using the `virtualenv` module, which explicitly does not support non-Python.org or Microsoft Store Python builds on Windows: https://virtualenv.pypa.io/en/latest/installation.html#windows

I don't think there's anything for Poetry to do here -- if you want to see this change, you are best off petitioning upstream in virtualenv.
